### DISCUSSION TITLE
Applying page selection on *.pdf

### DISCUSSION THREAD

**Discussion by Moghi** [qpdf/qpdf#538] [Q&A]
Hi, 
First of all thank you for qpdf, what a useful  tool!

I am using qpdf to merge all pdf files in a directory and I would like to merge only the first page of multiple inputfiles. According to the [qpdf documentation on page-selection](http://qpdf.sourceforge.net/files/qpdf-manual.html#ref.page-selection),  this should be possible. I have tried couple variants without luck:
```
qpdf --empty --pages *.pdf 1-1 -- "output.pdf"
qpdf --empty --pages *.pdf 1 -- "output.pdf"
```

Is it possible that page-selection is not applied to the results of `*.pdf`?

**Comment by jberkenbilt** [comment]
This isn't really a qpdf issue. It's really a shell issue. qpdf never sees `*.pdf` since the shell expands this and replaces it with all the files on the command line, so you are actually running `qpdf --empty --pages file1.pdf file2.pdf file3.pdf .. 1 -- output.pdf`, which is why the selector only applies to the last file. On Windows, you get identical behavior even though the wildcard substitution is performed a little later.

If you are using Linux, MacOS, or another UNIX-like environment,  you can run

```
qpdf --empty --pages $(for i in *.pdf; do echo $i 1; done) -- output.pdf
```

If you are in Windows, the above would work in bash. I don't know how to do it in powershell or cmd, but it would be easy to script.

**Comment by Moghi** [comment]
thank you for the answer. Now I get my error, I mistakenly expected the page option to be applied to all of the input files.
Yes I am using qpdf on Linux and MacOS, so I appreciate the example. I verified it and it works as expected.
Do you think it would be a plausible request to add this functionality to qpdf for ease of use?

**Reply by jberkenbilt** [reply]
Unfortunately, it's not possible to make this change as it would be a major non-compatible change to the CLI. I have a hard rule against changes that make something that has been previously valid remain valid but mean something else. If we wanted to have something that meant to apply a specific page selector to every specified input file, there would have to be a separate syntax for it, and I'm disinclined to do that since 1) there is a simple workaround, and 2) qpdf's CLI syntax is already way too complicated. I appreciate the suggestion though -- it's always useful to hear what people would find helpful. :-)

**Reply by Moghi** [reply]
Thank you for taking the time to answer. I definetly get your standpoint, and totally agree with you. What I don't understand is why I took so long to answer you (Maybe I wanted to spare you the reading effort). Thank you for maintaining qpdf.
